app/apps/views.js

```diff
@@ -148,3 +151,28 @@ exports.app_details = function (req, res) {
 
   });
 };
+
+
+exports.app_edit = [
+  ensureLoggedIn('/login'),
+  function(req, res, next) {
+    // This is making three requests cuncurrently to get the app details,
+    // the list of bucket and the list of users.
+    // If all of them succeed it then renders `apps/edit.html`
+    let app_request = api.get_app(req.params.id);
+    let buckets_request = api.list_buckets();
+    let users_request = api.list_users();
+
+    Bluebird
+      .all([app_request, buckets_request, users_request])
+      .spread(function(app_response, buckets_response, users_response) {
```

Comment:
Bluebird provides a non-standard Promises implementation. This can be achieved with the standard Promises API:
```javascript
Promise
  .all([app_request, buckets_request, users_request])
  .then((responses) => {
    let [app_response, buckets_response, users_response] = responses;
    ...
```